File: common/src/main/java/com/ultreon/mods/advanceddebug/util/MathHelper.java
package com.ultreon.mods.advanceddebug.util;

import java.math.BigDecimal;

@SuppressWarnings("unused")
public final class MathHelper {
    private MathHelper() {
        throw new UnsupportedOperationException("Cannot instantiate a utility class");
    }

    public static int getDecimalPlaces(Float d) {
        String s = d.toString();
        if (s.endsWith(".0")) {
            return 0;
        }
        String[] split = s.split("\\.");
        if (split.length == 1) {
            return 0;
        }

        return split[1].length();
    }

    public static int getDecimalPlaces(Double d) {
        String s = d.toString();
        if (s.endsWith(".0")) {
            return 0;
        }
        String[] split = s.split("\\.");
        if (split.length == 1) {
            return 0;
        }

        return split[1].length();
    }

    public static int getDecimalPlaces(BigDecimal d) {
        String s = d.toString();
        if (s.endsWith(".0")) {
            return 0;
        }
        String[] split = s.split("\\.");
        if (split.length == 1) {
            return 0;
        }

        return split[1].length();
    }
}

File: common/src/main/kotlin/com/ultreon/mods/advanceddebug/util/MathHelper.kt
package com.ultreon.mods.advanceddebug.util

import java.math.BigDecimal

@Suppress("unused")
class MathHelper private constructor() {
    init {
        throw UnsupportedOperationException("Cannot instantiate a utility class")
    }

    companion object {
        fun getDecimalPlaces(d: Float): Int {
            val s = d.toString()
            if (s.endsWith(".0")) {
                return 0
            }
            val split = s.split("\\.".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
            return if (split.size == 1) {
                0
            } else split[1].length
        }

        fun getDecimalPlaces(d: Double): Int {
            val s = d.toString()
            if (s.endsWith(".0")) {
                return 0
            }
            val split = s.split("\\.".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
            return if (split.size == 1) {
                0
            } else split[1].length
        }

        fun getDecimalPlaces(d: BigDecimal): Int {
            val s = d.toString()
            if (s.endsWith(".0")) {
                return 0
            }
            val split = s.split("\\.".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
            return if (split.size == 1) {
                0
            } else split[1].length
        }
    }
}